### PR TITLE
Research kit on fhir

### PR DESCRIPTION
**Pull request by ShaunDonn**

### CODE REVIEW

File: ResearchKitOnFhir/researchKitOnFhir/ViewControllers/SurveyListViewController.swift [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    var smartClient: Client?
+    var didAttemptAuthentication = false
+    static var questionnaireList = [Int: QuestionnaireType]()
+    var todoQList = [QuestionnaireType]()
+    var completeQList = [QuestionnaireType]()
+    let tableView = UITableView()
+    static var QList = [QListCategory]()
+    
+    struct Cells {
+        static let buttonCell = "buttonCell"
```
Comment (namalu) [inline ResearchKitOnFhir/researchKitOnFhir/ViewControllers/SurveyListViewController.swift:21]: You can also add a static property to the CustomTableViewCell for the reuse identifier. It's a little bit cleaner to have everything related to that cell in the cell class itself.

**Review by namalu — DISMISSED** [review]

File: ResearchKitOnFhir/researchKitOnFhir/README.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+| dateTime          | ORKDateAnswerFormat                                                               | style: ORKDateAnswerStyle.dateAndTime                                                                                                     |
+
+**Threshold of 10 arbitrarily set in func MultipleChoiceFormat in FHIRtoRK.swift: if all responses are shorter than 10 characters, Value Picker format is displayed. Otherwise, TextChoice format is displayed.
+
+
+## Authentication and Configuration
+ResearchKitOnFhir uses [SMART on FHIR](https://docs.smarthealthit.org) to integrate the app with a FHIR® Server. Current implementation supports configuration through the Config.json file, which the user populates with their FHIR® Server URL, SMART Client ID, and Patient ID, corresponding to ID of [Patient Resource](https://www.hl7.org/fhir/patient.html#resource) linked to intended app user. For example, a Patient who is accessed in the FHIR® Server through the URI "Patient/samplePatientName" will require a Config.json file with:
+
+```
+"patientId": "samplePatientName"
```
[the remark this answers is not in the document]
Comment (ShaunDonn) [reply ResearchKitOnFhir/researchKitOnFhir/README.md:30]: This is the id of the Patient that shows up in the uri for that Patient when querying the server; it's not a field in the Patient resource. 

It comes from this field in a Task resource: "task.basedOn?[0].reference?.string" which is called in populateQuestionnaireList() in SurveyListViewController.swift

Is this called a name?

File: ResearchKitOnFhir/researchKitOnFhir/ViewControllers/LandingScreenViewController.swift [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    
+    var didAttemptAuthentication = false
+    
+    @IBOutlet weak var authenticatedPatientButton: UIButton!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        #if targetEnvironment(simulator)
+        configMessageLabel.text = "Drag the Config.json file onto the Simulator screen to begin."
```
[the remark this answers is not in the document]
Comment (ShaunDonn) [reply ResearchKitOnFhir/researchKitOnFhir/ViewControllers/LandingScreenViewController.swift:21]: i.e. "Loading your configuration"?

File: ResearchKitOnFhir/researchKitOnFhir/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (namalu) [reply ResearchKitOnFhir/researchKitOnFhir/README.md:30]: If it is a reference, it sounds like the FHIR id. An example would of a FHIR id would be the guid in this path:

```
https://{{FHIR_SERVER}}/Patient/a76c6a04-b35b-4b77-9559-0c3da206aa2a
```

File: ResearchKitOnFhir/researchKitOnFhir/ViewControllers/LandingScreenViewController.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (namalu) [reply ResearchKitOnFhir/researchKitOnFhir/ViewControllers/LandingScreenViewController.swift:21]: Maybe:

"Tap on your configuration file and choose this application to open it."

**Review by namalu — APPROVED** [review]

**Pull request merged**